src/adam_test_data/main.py

```diff
@@ -177,23 +177,90 @@ def write_sorcha_inputs(
 
     if format == "csv":
-        sep = ","
+        delimiter = ","
     elif format == "whitespace":
-        sep = " "
+        delimiter = " "
     else:
         raise ValueError("format must be either 'csv' or 'whitespace'")
 
-    orbits_df.to_csv(orbit_file, index=False, sep=sep, float_format="%.15f")
-    properties_df.to_csv(properties_file, index=False, sep=sep, float_format="%.15f")
+    # Normally, it should be as easy as this:
+    # pa.csv.write_csv(
+    #     orbits_table,
+    #     orbit_file,
+    #     write_options=pa.csv.WriteOptions(
+    #         include_header=True, delimiter=delimiter, quoting_style="none"
+    #     ),
+    # )
+    # pa.csv.write_csv(
+    #     properties_table,
+    #     properties_file,
+    #     write_options=pa.csv.WriteOptions(
+    #         include_header=True,
+    #         delimiter=delimiter,
+    #         quoting_style="none",
+    #     ),
+    # )
+    # But sorcha doesn't know how to handle quotes in the header
+    # and the quoting_style="none" only applies to the row data and
+    # not the header. That is, the header names will remain surrounded
+    # by double quotes.
+    # See: https://github.com/apache/arrow/issues/41239
+
+    # Instead we write the CSVs to a string buffer
+    # and then strip the quotes from the header.
+    #
+    # Nevermind, this also doesn't work because object IDs can have
+    # "structural characters". Here is an example error if you
+    # set quoting_style="none":
+    # CSV values may not contain structural characters
+    # if quoting style is "None". See RFC4180. Invalid value: (2013 RR165)
+    #
+    # Furthermore, sorcha can't handle quotes in the data either...
+    # Instead we use the "needed" quoting style which adds quotes to the
+    # row data and the header. We then strip the quotes from everything with
+    # a replace call.
+    with pa.BufferOutputStream() as stream:
+        pa.csv.write_csv(
+            orbits_table,
+            stream,
+            write_options=pa.csv.WriteOptions(
+                include_header=True, delimiter=delimiter, quoting_style="needed"
+            ),
+        )
+        orbits_csv = stream.getvalue().to_pybytes().decode("utf-8")
+
+        # Strip the quotes from the header and body
+        orbits_csv = orbits_csv.replace('"', "")
+
+        with open(orbit_file, "w") as f:
+            f.write(orbits_csv)
+
+    with pa.BufferOutputStream() as stream:
+        pa.csv.write_csv(
+            properties_table,
+            stream,
+            write_options=pa.csv.WriteOptions(
+                include_header=True,
+                delimiter=delimiter,
+                quoting_style="needed",
+            ),
+        )
+        properties_csv = stream.getvalue().to_pybytes().decode("utf-8")
+
+        # Strip the quotes from the header and body
+        properties_csv = properties_csv.replace('"', "")
+
+        with open(properties_file, "w") as f:
+            f.write(properties_csv)
```

Comment:
@akoumjian The adventure into using pyarrow's CSV writer did not go as planned.

Reply:
Hm. Less than ideal as the entire csv needs to live in memory for a bit and it could be an order of magnitude larger than the pyarrow table. I would almost be inclined to just do a streaming rewrite on disk.

```python
import os

def remove_quotes(file_path):
    # Create a temporary file path
    temp_file_path = file_path + '.tmp'

    # Open the original file for reading in binary mode
    # and the temporary file for writing in binary mode
    with open(file_path, 'rb') as infile, open(temp_file_path, 'wb') as outfile:
        while True:
            # Read a chunk of data (e.g., 64KB)
            chunk = infile.read(65536)
            if not chunk:
                break
            # Remove quote characters
            chunk = chunk.replace(b'"', b'')
            # Write the processed chunk to the temporary file
            outfile.write(chunk)

    # Replace the original file with the temporary file
    os.replace(temp_file_path, file_path)

# Example usage:
remove_quotes('your_file.txt')
```
Since we are replacing simple ASCII quote characters, we should be able to read in binary mode, which is pretty fast. Using os.replace to swap in the modified file should also be fast and atomic. So I would just fun this function after writing each file to disk instead of to a stream.

Reply:
Done!